### PR TITLE
tidb: improve session.go and txn.go test coverage

### PR DESCRIPTION
**Pull request by tiancaiamao**
Try to cover more code...
@shenli @coocood

### CODE REVIEW

**Comment by shenli** [comment]
How about the coverage now?

**Comment by tiancaiamao** [comment]
Before: 
```
➜  tidb git:(master) ✗ go test -cover .                                                        
ok  	github.com/pingcap/tidb	106.867s	coverage: 76.0% of statements
```

After:
```
➜  tidb git:(coverage) ✗ go test -cover .     
ok  	github.com/pingcap/tidb	103.737s	coverage: 81.4% of statements
```

File: expression/builtin_string_test.go [hunk 1]
```diff
@@ -1101,19 +1101,12 @@ func (s *testEvaluatorSuite) TestChar(c *C) {
 		}
 	}
 
-	v := struct {
-		str    string
-		iNum   int64
-		fNum   interface{}
-		result string
-	}{"65", 66, nil, "AB"}
```
Comment (coocood) [inline expression/builtin_string_test.go]: Why remove this?

Comment (tiancaiamao) [reply expression/builtin_string_test.go]: I don't remove it, just modify it, to make go lint happy.

**Comment by coocood** [comment]
LGTM

**Pull request merged**